packer/builder/azure/arm/deployment_factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"encoding/json"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources"
+)
+
+type DeploymentFactory struct {
+	template string
+}
+
+func NewDeploymentFactory(template string) DeploymentFactory {
+	return DeploymentFactory{
+		template: template,
+	}
+}
+
+func (f *DeploymentFactory) create(templateParameters TemplateParameters) (*resources.Deployment, error) {
```

Comment:
what's with the unexported funcs? I know the distinction doesn't make a lot of sense since it's use will probably be with in the same package and it's a bit hard to see how it's will used eventually in this PR, but I think it's still a good idea to signal methods used outside of the core functionality of the struct by exporting those?